src/pytest_cases/case_parametrizer_new.py

```diff
@@ -265,9 +266,18 @@ def get_all_cases(parametrization_target,  # type: Callable
 
         filters += (filter,)
 
+    # Validate that we have a parametrization target if required for retrieving cases
+    if parametrize_with_cases is None:
+        if any(c is AUTO or c is THIS_MODULE or isinstance(c, str) for c in cases):
+            raise ValueError(
+                "Cases beginning with '.' or using AUTO require a parametrization target,"
+                " please use `get_all_cases(target_func, cases=...)`"
+            )
+
     # parent package
-    caller_module_name = getattr(parametrization_target, '__module__', None)
-    parent_pkg_name = '.'.join(caller_module_name.split('.')[:-1]) if caller_module_name is not None else None
+    if parametrization_target is not None:
+        caller_module_name = getattr(parametrization_target, '__module__', None)
+        parent_pkg_name = '.'.join(caller_module_name.split('.')[:-1]) if caller_module_name is not None else None
```

Comment:
Could this be merged with the above branch of the if ?

Also, if `parametrization_target` is not used at all below this point and only `caller_module_name` / `parent_pkg_name` are, we could imagine to accept both 

 - None (new default as per your proposal)
 - a module name (in case you wish to pre-collect various cases from an entire module, in order to resolve the relative imports and AUTO mode). Note: I may be too optimistic here, maybe a module name is not enough. An alternative is to accept a module object
 - a parametrization target

Finally, when None is passed, we could rely on the `get_caller_module` utility function in order to grab the calling module (`from .fixture__creation import get_caller_module`).

Reply:
Yes it could be merged, just made into an `else` statement. I was thinking to put it into the for loop but it made more sense to catch the error early to me.

I'm not sure I understand the bullet points but given the last statement of using `get_caller_module`, how does this behaviour sound

```python
# The current behaviour
def get_all_cases(f, cases=...)


# The functionality implemeneted
def get_all_cases(cases=[case_1, ..., case_n])

# Change this from being an error to being valid.
# Would use `get_caller_module` so these args work as expected
def get_all_cases(cases='.')
def get_all_cases(cases=AUTO)
def get_all_cases(cases='.xyz')
```

With respect to:
> a module name (in case you wish to pre-collect various cases from an entire module, in order to resolve the relative imports and AUTO mode). Note: I may be too optimistic here, maybe a module name is not enough. An alternative is to accept a module object

Would this functionality be solved with the `cases='.'` or `cases=AUTO` way of calling as illustrated in the last section of the code snippet above? If not, could you provide a code snippet of what args you would pass in and what the behaviour would be?